pyglossary/plugins/cc_cedict/__init__.py

```diff
@@ -0,0 +1,57 @@
+import re
+from formats_common import log
+from . import conv
+
+enable = True
+format = "CC-CEDICT"
+# this typo is part of the API used by PyGlossary; don't change it!
+extentions = (".u8",)
+
+entry_count_reg = re.compile(r"#! entries=(\d+)")
+class Reader:
+    def __init__(self, glos):
+        self._glos = glos
+        self.file = None
+        self.entries_left = 0
+
+    def open(self, filename, encoding="utf-8"):
+        if self.file is not None:
+            self.file.close()
+        self.file = open(filename, "r", encoding=encoding)
+        for line in self.file:
+            match = entry_count_reg.match(line)
+            if match is not None:
+                count = match.groups()[0]
+                self.total_entries = self.entries_left = int(count)
+                break
+        else:
+            raise RuntimeError("CC-CEDICT: could not find entry count")
+
+    def close(self):
+        if self.file is not None:
+            self.file.close()
+        self.file = None
+        self.entries_left = 0
+
+    def __len__(self):
+        if self.file is None:
+            raise RuntimeError("CC-CEDICT: len(reader) called "\
+                    "while reader is not open")
+        return self.total_entries
```

Comment:
Should set this to `total_entries=None` in __init__, and cleaner to check `total_entries is None` instead of `file is None`